### PR TITLE
Version 3.0

### PR DESCRIPTION
**Pull request by hotline-emu**

### CODE REVIEW

**Comment by hotline-emu** [comment]
PR is awaiting packagist availability of exceptions

File: src/Filter/UnsignedInt.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * @param bool  $allowNull Indicates if the value can be null.
+     * @param int   $minValue  Indicates the minimum acceptable value.
+     * @param int   $maxValue  Indicates the maximum acceptable value.
+     *
+     * @return int|null
+     *
+     * @throws \InvalidArgumentException if minvalue is not greater or equal to zero.
+     * @throws FilterException on failure to filter.
+     */
+    public static function filter($value, bool $allowNull = false, int $minValue = null, int $maxValue = PHP_INT_MAX)
```
Comment (chadicus) [inline src/Filter/UnsignedInt.php]: default `$minValue` to `0` instead of '`null`

File: src/Filter/UnsignedInt.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        if ($minValue === null) {
```
Comment (chadicus) [inline src/Filter/UnsignedInt.php]: If you default `$minValue` to `0`, this check can be removed

File: src/Filter/UnsignedInt.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            $minValue = 0;
+        } elseif (is_int($minValue) && $minValue < 0) {
```
Comment (chadicus) [inline src/Filter/UnsignedInt.php]: you can remove the `is_int` check

**Review by chadicus — APPROVED** [review]

**Pull request merged**